flink-staging/flink-table/src/main/scala/org/apache/flink/api/table/codegen/ExpressionCodeGenerator.scala

```diff
@@ -489,6 +489,25 @@ abstract class ExpressionCodeGenerator[R](
             """.stripMargin
         }
 
+      case NumericIsNotNull(child) =>
+        val childCode = generateExpression(child)
+        if (nullCheck) {
+          childCode.code +
+            s"""
+               |boolean $nullTerm = ${childCode.nullTerm};
+               |if ($nullTerm) {
+               |  0;
+               |} else {
+               |  $resultTpe $resultTerm = Boolean.compare((${childCode.resultTerm} != null),false);
```

Comment:
Boolean.compare() could be problematic because it is only guaranteed to return a value smaller than 0 or larger then 0 if the two values are not equal.

What about?
```
$resultTpe $resultTerm = ${childCode.resultTerm} != null ? 1 : 0;